src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs

```diff
@@ -476,11 +476,20 @@ private void RecordBuildDependencies(string goListOutput, ISingleFileComponentRe
             GoComponent goComponent;
             if (dependency.Replace?.Path != null && dependency.Replace.Version != null)
             {
-                var dependencyReplacementName = $"{dependency.Replace.Path} {dependency.Replace.Version}";
-                goComponent = new GoComponent(dependency.Replace.Path, dependency.Replace.Version);
-                this.Logger.LogInformation("go Module {GoModule} being replaced with module {GoModuleReplacement}", dependencyName, dependencyReplacementName);
-                record.GoModPathAndVersion = dependencyName;
-                record.GoModReplacement = dependencyReplacementName;
+                try
+                {
+                    var dependencyReplacementName = $"{dependency.Replace.Path} {dependency.Replace.Version}";
+                    goComponent = new GoComponent(dependency.Replace.Path, dependency.Replace.Version);
+                    this.Logger.LogInformation("go Module {GoModule} being replaced with module {GoModuleReplacement}", dependencyName, dependencyReplacementName);
+                    record.GoModPathAndVersion = dependencyName;
+                    record.GoModReplacement = dependencyReplacementName;
+                }
+                catch (Exception ex)
+                {
+                    record.ExceptionMessage = ex.Message;
+                    this.Logger.LogWarning("tried to use replace module but got this error {ErrorMessage} using original module instead", ex.Message);
```

Comment:
Let's be explicit here on what is the original module and version we used, the replace version that failed, and the exception message.